src/utils/details.js

```diff
@@ -36,45 +38,49 @@ function createDetails ({
   secret,
   data
 }) {
+  const token = cryptoHelper.getPskToken()
+  const paymentKey = cryptoHelper.getPaymentKey(secret, token)
+
   const privateRequest = _createRequest({
-    method: 'PRIVATE',
+    statusLine: false,
     headers,
     data
   })
 
-  const encrypted = cryptoHelper.aesEncryptBuffer(privateRequest, secret)
+  const encrypted = cryptoHelper.aesEncryptBuffer(privateRequest, paymentKey)
   const publicRequest = _createRequest({
-    method: 'PUBLIC',
-    headers: unsafeHeaders,
+    statusLine: true,
+    headers: Object.assign({
+      'Key': 'hmac-sha-256 ' + base64url(token)
+    }, unsafeHeaders),
```

Comment:
If you set a `Key` header you would make the packet unreadable because this won't actually respect the Key you set for encrypting the packet and the recipient won't know how to generate the decryption key. Either it shouldn't allow you to set the `Key` header or it should use the token you provide to generate the payment key